src/azureFunction/azureFunctionUtils.ts

```diff
@@ -229,7 +229,7 @@ export function waitForNewFunctionFile(projectFile: string): Promise<string> {
 		const watcher = vscode.workspace.createFileSystemWatcher((
 			path.dirname(projectFile), '**/*.cs'), false, true, true);
 		const timeout = setTimeout(async () => {
-			reject(new Error(constants.timeoutError));
+			reject(constants.timeoutSqlBindingError);
```

Comment:
If I do `new Error...` then this will output two error messages as you see currently

Reply:
Well that's likely a bug in our error handling then. Did you look at where the second error dialog is coming from? I suppose it could be something from VS Code but that'd be pretty odd behavior to me.

Reply:
This does come from VS Code as far as I see since I searched for the strings and did not find them in mssql repo.

Reply:
Not the "Error running command" one - that comes from VS Code yes. I'm saying the other one which just has the error message that you were rejecting with.

Reply:
As far as I know. `reject(new Error...)` would be the same as `throw new Error...`. So that's why we get one from VS Code as mentioned and then another one (which we do not need anymore).
https://javascript.info/promise-error-handling#implicit-try-catch

Reply:
So throwing directly vs rejecting is the same - but you should be throwing an Error regardless. Not throwing a string.

What isn't clear to me is why this fixes the duplicate error messages. Because unless VS Code has something built in I don't see why throwing an Error object vs a string would cause an extra dialog to show like this - I've never seen that before and we throw errors everywhere else. Are you sure we aren't catching the error and displaying the additional message ourselves? That seems much more likely.